GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -833,17 +769,69 @@ protected override void DisposeCustomResources()
 
         private sealed class ItemTpComparer : IComparer<string>
         {
+            private readonly string _currentBranch;
             private readonly IDictionary<string, int> _orderDict;
             private readonly string _prefix;
+            private readonly Regex _remoteMasterRegex = new Regex("^.*/.*/master[^/]*$",
+                RegexOptions.Compiled | RegexOptions.CultureInvariant);
 
-            public ItemTpComparer(IDictionary<string, int> orderDict, string prefix)
+            public ItemTpComparer(IDictionary<string, int> orderDict, string prefix, string currentBranch = null)
             {
                 _orderDict = orderDict;
                 _prefix = prefix;
+                _currentBranch = currentBranch;
             }
 
             public int Compare(string a, string b)
             {
+                if (_currentBranch != null)
+                {
+                    if (a == _currentBranch)
+                    {
+                        return -1;
+                    }
+
+                    if (b == _currentBranch)
+                    {
+                        return 1;
+                    }
+                }
+
+                int defaultCompareResult = Comparer<string>.Default.Compare(a, b);
+
+                bool isLocalMaster_a = IsLocalMaster(a);
+                bool isLocalMaster_b = IsLocalMaster(b);
+                if (isLocalMaster_a && isLocalMaster_b)
+                {
+                    return defaultCompareResult;
+                }
+                else if (isLocalMaster_a)
+                {
+                    return -1;
+                }
+                else if (isLocalMaster_b)
+                {
+                    return 1;
+                }
+
+                bool isRemoteMaster_a = IsRemoteMaster(a);
+                bool isRemoteMaster_b = IsRemoteMaster(b);
+                if (isRemoteMaster_a && isRemoteMaster_b)
+                {
+                    return defaultCompareResult;
+                }
+                else if (isRemoteMaster_a)
+                {
+                    return -1;
+                }
+                else if (isRemoteMaster_b)
+                {
+                    return 1;
+                }
+
+#if true        // As long as GetSortedRefs() doesn't return the branches sorted by date, sort other branches alphabetically.
+                return defaultCompareResult;
```

Comment:
Discussion: I wish to see local branches before remote and (especially on GH where I have many remotes) origin-upstream-other remotes after that. But this PR is an improvement as-is.

Reply:
If you prefix the names of the other remotes with "z", you'll get most of this sorting - also in the left panel. Have a look at the screenshots.
Further tuning of the sorting should be addressed in conjunction with the git(?) bug #7181.

Reply:
> If you prefix the names of the other remotes

Kind of what I use now.
With more information, more requests to format it will come. But I got the master* change, so I am satisfied